### PR TITLE
Adds pull request, feature request and bug report templates.

### LINKED ISSUE #57: Create a GH Pull request template
We need to create a PR template for this project. This can be something similar to the one found in the AWS provider of cluster-api

https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/master/.github

Good first issue.

### LINKED ISSUE #58: Create GH Issue Template
We need to create an Issue template for this project. This can be something similar to the one found in the AWS provider of cluster-api

https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/master/.github

Good first issue.

### PR DESCRIPTION
**Pull request by ilhan-mstf**
Fixes #58, Fixes #57

### CODE REVIEW

**Comment by ilhan-mstf** [comment]
Contributor License Agreement is signed.

**Comment by ilhan-mstf** [comment]
Contributor License Agreement is signed.

**Comment by tariq1890** [comment]
@ilhan-mstf Thanks for your contribution. Can you make sure that the git config email matches the one you've used while signing the CLA?

**Comment by tariq1890** [comment]
Can you try this ?

`git commit --amend --author="Author Name <email@address.com>"`

and then force push ?

**Comment by ilhan-mstf** [comment]
@tariq1890 I missed the HelloSign point, now I guess I've completed signing process.

**Comment by tariq1890** [comment]
deferring to @justaugustus for final review.

File: .github/ISSUE_TEMPLATE/feature_request_template.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+[A clear and concise description of what you want to happen.]
+
+
+**Anything else you would like to add:**
+[Miscellaneous information that will assist in solving the issue.]
+
+
+**Environment:**
+
+- Cluster-api-provider-aws version:
```
Comment (marwanad) [inline .github/ISSUE_TEMPLATE/feature_request_template.md]: azure*

**Comment by tariq1890** [comment]
Whoops! Thanks for catching that Marwan!

File: .github/ISSUE_TEMPLATE/bug_report_template.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+**What did you expect to happen:**
+
+
+**Anything else you would like to add:**
+[Miscellaneous information that will assist in solving the issue.]
+
+
+**Environment:**
+
+- Cluster-api-provider-aws version:
```
Comment (justaugustus) [inline .github/ISSUE_TEMPLATE/bug_report_template.md]:
```suggestion
- cluster-api-provider-azure version: 
```

File: .github/ISSUE_TEMPLATE/feature_request_template.md (hunk 1 above)
Comment (justaugustus) [inline .github/ISSUE_TEMPLATE/feature_request_template.md]:
```suggestion
- cluster-api-provider-azure version: 
```

**Pull request merged**